include/zephyr/kernel.h

```diff
@@ -3602,6 +3602,14 @@ bool k_work_cancel_sync(struct k_work *work, struct k_work_sync *sync);
  */
 void k_work_queue_init(struct k_work_q *queue);
 
+/** @brief Run work queue using calling thread
+ *
+ * This will run the work queue forever, the function never returns
+ *
+ * @param queue the queue to run
```

Comment:
Missing doxygen for the configuration parameter.

Wouldn't it be possible to call `k_work_queue_stop` to make this function return?

Reply:
@pdgendt Needed one small fix I missed for using the `thread_id`, along with a test for it.